superset/assets/javascripts/dashboard/components/GridCell.jsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable react/no-danger */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SliceHeader from './SliceHeader';
+import ChartContainer from '../../chart/ChartContainer';
+
+import '../../../stylesheets/dashboard.css';
+
+const propTypes = {
+  timeout: PropTypes.number,
+  datasource: PropTypes.object,
+  isLoading: PropTypes.bool,
+  isExpanded: PropTypes.bool,
+  widgetHeight: PropTypes.number,
+  widgetWidth: PropTypes.number,
+  fetchSlice: PropTypes.func,
+  removeSlice: PropTypes.func.isRequired,
+  updateSliceName: PropTypes.func,
+  toggleExpandSlice: PropTypes.func,
+  slice: PropTypes.object,
+  chartKey: PropTypes.string,
+  formData: PropTypes.object,
+  filters: PropTypes.object,
+  addFilter: PropTypes.func,
+  clearFilter: PropTypes.func,
+  removeFilter: PropTypes.func,
+  exploreChartUrl: PropTypes.string,
+  exportCSVUrl: PropTypes.string,
+};
+
+class GridCell extends React.PureComponent {
+  getDescriptionId(slice) {
+    return 'description_' + slice.slice_id;
+  }
+
+  getHeaderId(slice) {
+    return 'header_' + slice.slice_id;
+  }
+
+  width() {
+    return this.props.widgetWidth - 10;
+  }
+
+  height(slice) {
+    const widgetHeight = this.props.widgetHeight;
+    const headerId = this.getHeaderId(slice);
+    const descriptionId = this.getDescriptionId(slice);
+    const headerHeight = this.refs[headerId] ? this.refs[headerId].offsetHeight : 30;
+    let descriptionHeight = 0;
+    if (this.props.isExpanded && this.refs[descriptionId]) {
+      descriptionHeight = this.refs[descriptionId].offsetHeight + 10;
+    }
+    return widgetHeight - headerHeight - descriptionHeight;
+  }
+
+  render() {
+    const slice = this.props.slice;
+    return (
+      <div className="slice-cell" id={`${slice.slice_id}-cell`}>
+        <div ref={this.getHeaderId(slice)}>
+          <SliceHeader
```

Comment:
So we have 2 types of chart headers (one in explore and one in dashboard) and eventually I think that should live inside `<Chart/>` as opposed to in those respective apps.

I'm thinking we'd add a `showHeader` bool prop to `Chart` as well as and `showExtendedHeader` or something like that.

We can save that for a future refactor if you want.

Reply:
It's hard to find common functions or common UI between slice header in dashboard and chart header in explore view :(
I assume we will have a UI/UX redesign very soon. we can provide different header in different `mode` together with redesign.